### PR TITLE
Log more details when we get the 0-length error

### PR DESCRIPTION
**Pull request by dims**
Change-Id: I9733521a3afd91c54c19fe08d0622df4187c0c0c

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

Add more available info when we log the error about `0-length response`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/rest/request.go [hunk 1]
```diff
@@ -1100,7 +1100,8 @@ func (r Result) Into(obj runtime.Object) error {
 		return fmt.Errorf("serializer for %s doesn't exist", r.contentType)
 	}
 	if len(r.body) == 0 {
-		return fmt.Errorf("0-length response")
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/rest/request.go]: There's no err or we'd have exited on line 1097

Comment (dims) [reply staging/src/k8s.io/client-go/rest/request.go]: fixed

**Pull request merged**